### PR TITLE
Add dotty to cross-build

### PR DESCRIPTION
**Pull request by Philippus**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -24,11 +34,20 @@ scalacOptions in (Compile, console) ~= (_ filterNot Set(
 unmanagedSourceDirectories in Compile += {
   val sourceDir = (sourceDirectory in Compile).value
   CrossVersion.partialVersion(scalaVersion.value) match {
-    case Some((2, n)) if n >= 13 => sourceDir / "scala-2.13+"
-    case _                       => sourceDir / "scala-2.13-"
+    case Some((2, n)) if n >= 13         => sourceDir / "scala-2.13+"
+    case Some((n, _)) if n == 0 | n == 3 => sourceDir / "scala-2.13+"
```
Comment (SethTisue) [inline build.sbt:38]: style nit: you can `case Some((0 | 3, _))`

**Review by SethTisue — COMMENTED** [review]
why was it necessary/desired to split `src/test` by Scala version?

**Comment by Philippus** [comment]
> cool, thanks for working on this.
> 
> wondering why was it necessary/desired to split `src/test` by Scala version?

Two reasons:
- dotty requires `import scala.language.implicitConversions`, if I add it for scala 2.x that triggers an `Unused import`.
- the dotty version of scalatest trips over the `should be(...)`-construct with `a "should" clause may not appear inside an "in" clause`, so I changed that to `shouldBe(...)`.

**Comment by SethTisue** [comment]
I think it would be a lot better if we could avoid the split, since it's a maintenance hassle. Such source splits shouldn't be necessary except under unusual circumstances.

Re: `implicitConversions`, one workaround would be to add `-language:implicitConversions` to `scalacOptions` on Scala 2 only. I'd be okay with that actually, although it's a bigger hammer than strictly necessary. Another possibility would be to try to `@nowarn` it — is that possible with imports? Not sure. A third possibility would be to `-Wconf` or `@nowarn` it to not fail the build.

I'm not actually suggesting this here in this repo, but as a general matter, if it's necessary to resort to version-specific sources, it's often possible to confine the necessary version-specific parts to just a small trait, and then mix that trait into the downstream code, which can remain unspilt. I haven't thought about whether that's possible here (since I think it shouldn't be necessary).

**Comment by Philippus** [comment]
I agree with you. In the end I settled for being less strict on fatal errors in the `test` scope.

**Review by SethTisue — APPROVED** [review]

**Pull request merged**